pkg/apis/yunikorn.apache.org/v1alpha1/type.go

```diff
@@ -0,0 +1,71 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Application struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   ApplicationSpec   `json:"spec"`
+	Status ApplicationStatus `json:"status"`
+}
+
+//Spec part
+
+type ApplicationSpec struct {
+	MinMember         int32  `json:"minMember"`
+	Queue             string `json:"queue"`
+	MaxPendingSeconds int32  `json:"maxPendingSeconds,omitempty"`
+}
+
+//Status part
+
+type ApplicationStateType string
+
+const (
+	NewApplicationState ApplicationStateType = "iota"
```

Comment:
The string representation for New state should be "New". iota used here: https://github.com/apache/incubator-yunikorn-core/blob/f4ea4502a52e06e64e5abd0c47fedf2e08fe12e3/pkg/cache/application_state.go#L53-L61 is a  keyword to represent successive integer constants